### ISSUE TITLE
side-effect of lsyncd in real-time log aggregator

### ISSUE THREAD

**Issue by mehrdad2000** [lsyncd/lsyncd#691]
Hi
TL;DR: need log collector for file with minimum configuration, and support different conditions.

I have 3 servers that generate log file daily with size about 12GB (12*3=36GB)

How can I gather these files on centralize log server.

server1 >
server2 > centralize log
server3 >

FYI1: lsyncd, rsyslog, filebeat, syslog-ng, fluentd, ... are available solution but I can't decide which one is more reliable for this issue.

FYI2: raw data is important , and doesn't be missed. (don't want to clean data exact log files is an important factor for me)

FYI3: like splunk forwarder whenever servers or network down, after issue resolve it will continuously send data. (AFAIK rsyslog use tracker file when server stopped and try to send remain file after service start again but while service down new file create with different name and structure can't track) (but splunk forwarder can handle this situation even when forwarder service on servers down, whenever start it can discover any file on that path).
How lsyncd handle this condition? How lsyncd track files? how it behavior in case of  "centralize log" server down or has network issue between servers and centralize log? does it buffer or queue somewhere in servers side host? does it affect to TCP queue of servers OS?

FYI4: here is the path of my log /opt/log/*
different file, with different name may create here , I need to dynamically everything on this path send to the centralize log.

Any idea?

Thanks,

**Comment by poelzi** [comment]
lsync might not be the best solution for this. Log files are usually constantly written, sometimes even constant. lsync is more a tool to keep a source directory in sync with n target directories.
You could use lsync to sync your log folder into some target machine, but for logging, there are better solutions, you already mentioned. If you want to periodically sync this folder to some other machine, sure, you can do that. Give each server a dedicated target directory.
File transfers that fail are retried automatically.
